ompy/dist/fermi_dirac.py

```diff
@@ -0,0 +1,127 @@
+import numpy as np
+
+import theano.tensor as tt
+from pymc3.distributions.continuous import (
+    PositiveContinuous,
+    assert_negative_support,
+    draw_values,
+    generate_samples)
+from pymc3.distributions.dist_math import bound
+from pymc3.theanof import floatX
+
+
+class FermiDirac(PositiveContinuous):
+    """
+    Fermi-Dirac distribution.
+
+    The pdf of this distribution is
+
+    .. math::
+
+        f(x \mid \lambda \mu) =
+            \frac{\lambda}{\lambda\mu + \ln(1 + e^{-\lambda\mu})}
+            \frac{1}{e^{\lambda(x - \mu)} + 1}
+
+    .. plot::
+
+
+    ========  ============================
+    Support   :math:`x \in [0, \infty)`
+    Mean
+    Variance
+    ========  ============================
+
+
+    Parameters
+    ----------
+    lam: float
+        Rate of decay at mu (lam > 0)
+    mu: float
+        Decay position
+    """
+
+    def __init__(self, lam, mu, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.lam = tt.as_tensor_variable(floatX(lam))
+        self.mu = tt.as_tensor_variable(floatX(mu))
+        assert_negative_support(lam, "lam", "FermiDirac")
+
+        self.median = self._ppf(0.5, self.lam, self.mu)
+
+    def _ppf(self, q, lam, mu):
+        """
+        Calculate the CDF for the Fermi-Dirac distribution.
+        """
+        N = np.log(1 + np.exp(lam*mu))
+        return mu - np.log(np.exp(N*(1-q)) - 1)/lam
```

Comment:
I think this should be reformulated / simplified in the article to have the same notation. It's somewhat confusing if e.g. `N` there and here are not the same